narwhals/_dask/dataframe.py

```diff
@@ -113,12 +115,27 @@ def select(
         df = self._native_frame.assign(**new_series).loc[:, list(new_series.keys())]
         return self._from_native_frame(df)
 
-    def drop_nulls(self: Self, subset: str | list[str] | None) -> Self:
+    def drop_nulls(
+        self: Self, subset: str | Selector | list[Selector | str] | None
+    ) -> Self:
         if subset is None:
             return self._from_native_frame(self._native_frame.dropna())
-        subset = [subset] if isinstance(subset, str) else subset
+
+        from narwhals.selectors import Selector
+
         plx = self.__narwhals_namespace__()
-        return self.filter(~plx.any_horizontal(plx.col(*subset).is_null()))
+        subset_: Iterable[str | Selector] = (
+            [subset] if isinstance(subset, (str, Selector)) else subset
+        )
+
+        column_names = [c for c in subset_ if isinstance(c, str)]
+        selector_names = self.select(
+            *[c.is_null()._call(plx) for c in subset_ if isinstance(c, Selector)]
+        ).columns
```

Comment:
I am _giving for granted_ that the optimizer knows we are only getting the column names at the end